### DISCUSSION TITLE
Why so many kwargs?

### DISCUSSION THREAD

**Discussion by pohlt** [containers/podman-py#191] [General]
First of all, thanks for your work you put into this package. I do appreciate this. ❤️ 

While using it, I noticed that so many function use `**kwargs` which means that IDEs cannot support you with the parameters.
I'm sure there's a good reason for it, but I could not find it when I looked into the functions.

When I have to consult the implementation of a function (or the excellent API documentation) I might as well send the API request myself (and use an async request replacement like httpx, but that's another topic).

The same goes for some more complex parameters which are implemented as nested dicts and lists: Sure, it works, but there's no support in building those structures. Again, I have to look into the API documentation. Instead, dataclasses which know what parameters to expect would be more supportive and pythonic.

**Comment by rhatdan** [comment]
@mwhahaha @jwhonce PTAL
